cmd/profiles.go

```diff
@@ -97,40 +104,68 @@ User profiles are not overwritten on upgrade of elastic-stack, and can be freely
 				return errors.Wrap(err, "error listing all profiles")
 			}
 
-			table := tablewriter.NewWriter(os.Stdout)
-			var profilestable = profileToList(profileList)
-
-			table.SetHeader([]string{"Name", "Date Created", "User", "Version", "Path"})
-			table.SetHeaderColor(
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-			)
-			table.SetColumnColor(
-				twColor(tablewriter.Colors{tablewriter.Bold, tablewriter.FgCyanColor}),
-				tablewriter.Colors{},
-				tablewriter.Colors{},
-				tablewriter.Colors{},
-				tablewriter.Colors{},
-			)
-
-			table.SetAutoMergeCells(false)
-			table.SetRowLine(true)
-			table.AppendBulk(profilestable)
-			table.Render()
+			format, err := cmd.Flags().GetString(cobraext.ProfileFormatFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ProfileFromFlagName)
+			}
 
-			return nil
+			switch format {
+			case tableFormat:
+				return formatTable(profileList)
+			case jsonFormat:
+				return formatJSON(profileList)
+			default:
+				return fmt.Errorf("format %s not supported", format)
+			}
 		},
 	}
+	profileListCommand.Flags().String(cobraext.ProfileFormatFlagName, "table", cobraext.ProfileFormatFlagDescription)
 
 	profileCommand.AddCommand(profileNewCommand, profileDeleteCommand, profileListCommand)
 
 	return cobraext.NewCommand(profileCommand, cobraext.ContextGlobal)
 
 }
 
+func formatJSON(profileList []profile.Metadata) error {
+	data, err := json.Marshal(profileList)
+	if err != nil {
+		return errors.Wrap(err, "error listing all profiles in JSON format")
+	}
+
+	fmt.Print(string(data))
+
```

Comment:
I can remove it, although I'm tempted to keep it because I added it on purpose: I personally like to make a clear separation of one line before the return sentence. Exactly the same we see in the previous code right after the Render method.